python-package/lightgbm/sklearn.py

```diff
@@ -812,6 +861,16 @@ def predict(self, X, raw_score=False, start_iteration=0, num_iteration=None,
         ):
             predict_params.pop(alias, None)
         predict_params.update(kwargs)
+
+        # number of threads can have values with special meaning which is only applied
+        # in the scikit-learn interface, these should not reach the c++ side as-is
+        n_jobs = self.n_jobs
+        num_threads_aliases = _ConfigAliases.get("num_threads")
+        for alias in num_threads_aliases:
```

Comment:
Just to decrease the number of code lines without hurting the code readability.
```suggestion
        for alias in _ConfigAliases.get("num_threads"):
```